pytest_fxa/plugin.py

```diff
@@ -74,20 +87,21 @@ def fxa_account(fxa_client, fxa_email):
     password = ''.join([random.choice(string.ascii_letters) for i in range(8)])
     FxAccount = collections.namedtuple('FxAccount', 'email password')
     fxa_account = FxAccount(email=account.email, password=password)
-    session = fxa_client.create_account(fxa_account.email,
-                                        fxa_account.password)
-    logger.info('Created: {}'.format(fxa_account))
-    account.fetch()
-    message = account.wait_for_email(lambda m: 'x-verify-code' in m['headers'])
-    session.verify_email_code(message['headers']['x-verify-code'])
-    logger.info('Verified: {}'.format(fxa_account))
-    yield fxa_account
     try:
-        account.clear()
-        fxa_client.destroy_account(fxa_account.email, fxa_account.password)
-        logger.info('Removed: {}'.format(fxa_account))
-    except ClientError as e:
-        # 'Unknown Account' error is ok - account already deleted
-        # https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#response-format
-        if e.errno not in [102]:
-            raise
+        session = fxa_client.create_account(fxa_account.email,
+                                            fxa_account.password)
+        logger.info('Created: {}'.format(fxa_account))
+        account.fetch()
+        message = account.wait_for_email(
+            lambda m: 'x-verify-code' in m['headers'] and
+            session.uid == m['headers']['x-uid']
+        )
+        if message is None:
+            raise AssertionError(
+                'An empty message was returned from the server.'
```

Comment:
I'm okay with us raising an `AssertionError`, however we should instead use `assert message is not None, 'message'` instead of raising it directly. I also believe the message is inaccurate.. Isn't the issue that no matching message was recieved, rather than an empty message?

Reply:
I think it is both. ```wait_for_email``` will either return the message that says 'Click here to verify your email' or it will return None, which will throw a ```TypeError```.

Reply:
```verify_email_code``` is where the issue with not matching the code within the message vs the account is.

Reply:
```wait_for_email``` calls another method ```find_email``` if that fails it returns ```None```. I don't see how we can tell what failed when ```message == None```.